File: 3084-count-substrings-starting-and-ending-with-given-character/3084-count-substrings-starting-and-ending-with-given-character.java
class Solution {
    public long countSubstrings(String s, char c) {
        long c_count = 0,ans = 0;;   // Initilize long variable c_count and ans as 0.
        for(int i=s.length()-1;i>=0;i--){   // Loop through the string s and check if the ith character is equal to c, if so then increment c_count by 1 and add c_count to the numerator.
            if(s.charAt(i)==c){
                c_count++;
                ans = ans + c_count;
            }
        }
        return ans;   // return ans.
    }
}

File: 3084-count-substrings-starting-and-ending-with-given-character/3084-count-substrings-starting-and-ending-with-given-character.js
/**
 * @param {string} s
 * @param {character} c
 * @return {number}
 */
var countSubstrings = function(s, c) {
    let x = 0;
    for (let i = 0 ; i < s.length; i++) {
        if (s[i] === c) x++;
    }
    return (x * (x-1))/2 + x;
};